lib/subjects/selector.rb

```diff
@@ -115,14 +116,16 @@ def user_subject_queue
       end
     end
 
-    def find_subject_queue(user=queue_user)
+    def find_subject_queue
       SubjectQueue.by_set(subject_set_id)
-        .find_by(user: user, workflow: workflow)
+        .find_by(user: queue_user, workflow: workflow)
     end
 
-    def dequeue_for_logged_in_user(sms_ids)
+    def dequeue_ids(queue, sms_ids)
       if queue_user
-        DequeueSubjectQueueWorker.perform_async(workflow.id, sms_ids, queue_user.try(:id), subject_set_id)
+        queue.dequeue_update(sms_ids)
```

Comment:
Dequeue for logged in users is not in the request cycle. if this impacts too much we can move it to the background again.